pkg/storage/tsdb/block/fetcher.go

```diff
@@ -175,13 +175,27 @@ var (
 )
 
 // loadMeta returns metadata from object storage or error.
+// fetchAttributes indicates whether file attributes, such as lastModified, should be fetched and populated on the resulting meta.
 // It returns ErrorSyncMetaNotFound and ErrorSyncMetaCorrupted sentinel errors in those cases.
-func (f *MetaFetcher) loadMeta(ctx context.Context, id ulid.ULID) (*Meta, error) {
+func (f *MetaFetcher) loadMeta(ctx context.Context, id ulid.ULID, fetchAttributes bool) (meta *Meta, err error) {
 	var (
 		metaFile       = path.Join(id.String(), MetaFilename)
 		cachedBlockDir = filepath.Join(f.cacheDir, id.String())
 	)
 
+	if fetchAttributes {
```

Comment:
I think we misunderstood each other. I don't think we should be fetching attributes for ALL meta.json files. We can have tens of thousands meta.json files in one tenant, but most of these describe blocks that are no longer going to be compacted. We run meta-fetcher for tenant at each compaction cycle. That could trigger a lot of `GetAttributes` calls.

We should only fetch attributes for blocks that we're actually running compaction for. That means 1) in `jobWaitPeriodElapsed` as before (this PR doesn't need to change that function) and 2) after compaction job finished.